src/routes/route-factory.js

```diff
@@ -28,9 +29,14 @@ module.exports = function createRoute(repository) {
   }
 
   function list(request, response, next) {
-    repository.list()
+    repository.list(request.query)
       .then((result) => {
-        response.json(result);
+        response.json({
+          total: result.length,

```

Comment:
`result.length` may not be equal to `total`, especially when you set the limit and offset in sql query

Reply:
'limit' and 'offset' are query params from GET that are passed to the database

Reply:
we get 'result' from database, why would its length not be equal to the total of items that are returned?